modules/substrate/src/lib.rs

```diff
@@ -215,6 +191,28 @@ decl_module! {
 
 			Ok(())
 		}
+
+		/// Bootstrap the bridge pallet with an initial header and authority set from which to sync.
+		///
+		/// The initial configuration provided does not need to be the genesis header of the bridged
+		/// chain, it can be any arbirary header. You can also provide the next scheduled set change
+		/// if it is already know.
+		///
+		/// This function is only allowed to be called from a trusted origin and writes to storage
+		/// with practically no checks in terms of the validity of the data. It is important that
+		/// you ensure that valid data is being passed in.
+		// TODO: Update weights [#78]
+		#[weight = 0]
+		pub fn initialize(
+			origin,
+			header: BridgedHeader<T>,
+			authority_list: sp_finality_grandpa::AuthorityList,
+			set_id: sp_finality_grandpa::SetId,
+			scheduled_change: Option<ScheduledChange<BridgedBlockNumber<T>>>,
+		) {
+			let _ = ensure_root(origin)?;
```

Comment:
IMHO to prevent abuses this should only be callable if the bridge is not already initialized. IMHO resetting/re-initializing the state of the bridge should be a separate issue, cause it will most likely involve a bit of cleaning up as well to make sure things stay consistent.

Reply:
```suggestion
			let _ = ensure_owner_or_root(origin)?;
```
Also this should allow the owner, introduced in #472

Reply:
The owner stuff that was implemented in #472 is only for the message lane pallet. The Substrate pallet doesn't have that yet which is why I mentioned this:

> I'll add support for the pallet owner to also be able to initialize it in a different PR tomorrow.

As far as this goes: 

> IMHO to prevent abuses this should only be callable if the bridge is not already initialized. IMHO resetting/re-initializing the state of the bridge should be a separate issue, cause it will most likely involve a bit of cleaning up as well to make sure things stay consistent.

I can add a check in that other PR that this should only be called if the pallet hasn't been initialized.

Reply:
> The Substrate pallet doesn't have that yet which is why I mentioned this:

Oh, right. Sorry I missed that.